spec/rails_helper.rb

```diff
@@ -15,20 +15,63 @@
 
 # require "capybara/rails"
 # require 'capybara/rspec'
-require 'capybara/poltergeist'
+require 'selenium/webdriver'
 require 'rack_session_access/capybara'
 require 'capybara-screenshot/rspec'
 require 'database_cleaner'
 
 require 'rake'
 require 'rails/tasks'
 
-# Specs flagged with `js: true` will use Capybara's JS driver. Set
-# that JS driver to :poltergeist
-Capybara.register_driver :poltergeist do |app|
-  Capybara::Poltergeist::Driver.new(app, timeout: 1.minute, phantomjs_options: ['--ssl-protocol=TLSv1.2'])
+# Specs flagged with `js: true` will use Capybara's JS driver.
+Capybara.register_driver :headless_chrome do |app|
+  # set timeout to 60s http://www.testrisk.com/2016/05/change-default-timeout-and-wait-time-of.html
+  # need to use read_timeout and open_timeout https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/remote/http/default.rb
+  client = Selenium::WebDriver::Remote::Http::Default.new
+  client.read_timeout = 60
+  client.open_timeout = 60
+
+  # http://technopragmatica.blogspot.com/2017/10/switching-to-headless-chrome-for-rails_31.html
+  capabilities = Selenium::WebDriver::Remote::Capabilities.chrome(
+    # This makes javascript console logs available, but doesn't cause them to appear in real time
+    # to display javascript logs in the rspec output, add `puts page.driver.browser.manage.logs.get(:browser)`
+    # in the desired test location
+    loggingPrefs: { browser: 'ALL', client: 'ALL', driver: 'ALL', server: 'ALL' }
+  )
+
+  # disable-gpu option is temporarily necessary, possibly only for Windows
+  # https://developers.google.com/web/updates/2017/04/headless-chrome#cli
+  # no-sandbox was necessary for another application's Docker container for CI/CD
+  # https://about.gitlab.com/2017/12/19/moving-to-headless-chrome/
+  # https://developers.google.com/web/updates/2017/04/headless-chrome#faq
+  options = Selenium::WebDriver::Chrome::Options.new(
+    args: %w[headless disable-gpu no-sandbox --window-size=1440,900]
```

Comment:
I assume this is what we would change if we want taller screen shots?

Reply:
yes